packages/shared/watchListChanges/index.test.ts

```diff
@@ -0,0 +1,64 @@
+import { nextTick, ref } from 'vue-demi'
+import { watchListChanges } from '.'
+
+describe('watchListChanges', () => {
+  it('should work when two lists are different', async () => {
+    const spy = vitest.fn((newList, oldList, added, removed) => {
+      expect(newList).toEqual([1, 1, 4])
+      expect(oldList).toEqual([1, 2, 3])
+      expect(added).toEqual([1, 4])
+      expect(removed).toEqual([2, 3])
+    })
+
+    const num = ref([1, 2, 3])
+    watchListChanges(num, spy)
+    num.value = [1, 1, 4]
+    await nextTick()
+    expect(spy).toBeCalledTimes(1)
+  })
+
+  it('should work when two lists are identical', async () => {
+    const spy = vitest.fn((newList, oldList, added, removed) => {
+      expect(newList).toEqual([1, 2, 3])
+      expect(oldList).toEqual([1, 2, 3])
+      expect(added).toEqual([])
+      expect(removed).toEqual([])
+    })
+
+    const num = ref([1, 2, 3])
+    watchListChanges(num, spy)
+    num.value = [1, 2, 3]
+    await nextTick()
+    expect(spy).toBeCalledTimes(1)
+  })
+
+  it('should work with list push', async () => {
+    const spy = vitest.fn((newList, oldList, added, removed) => {
+      expect(newList).toEqual([1, 2, 3, 4])
+      expect(oldList).toEqual([1, 2, 3])
+      expect(added).toEqual([4])
+      expect(removed).toEqual([])
+    })
+
+    const num = ref([1, 2, 3])
+    watchListChanges(num, spy, { deep: true })
+    num.value.push(4)
+    await nextTick()
+    expect(spy).toBeCalledTimes(1)
+  })
+
+  it('should work with list splice', async () => {
+    const spy = vitest.fn((newList, oldList, added, removed) => {
+      expect(newList).toEqual([1, 5, 6, 7, 3])
+      expect(oldList).toEqual([1, 2, 3])
+      expect(added).toEqual([5, 6, 7])
+      expect(removed).toEqual([2])
+    })
+
+    const num = ref([1, 2, 3])
+    watchListChanges(num, spy, { deep: true })
+    num.value.splice(1, 1, 5, 6, 7)
+    await nextTick()
+    expect(spy).toBeCalledTimes(1)
```

Comment:
Would be great to also have some test for `reactive([1,2,3])` and `() => arr.value` as watch source.